### PR TITLE
Pass build mode-specific bytecode generation options to frontend_server.

### PR DESCRIPTION
**Pull request by rmacnak-google**
Fixes various Flutter framework test failures when bytecode is enabled.

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/build.dart [hunk 1]
```diff
@@ -308,7 +308,7 @@ class AOTSnapshotter {
       extraFrontEndOptions: extraFrontEndOptions,
       linkPlatformKernelIn: true,
       aot: true,
-      enableAsserts: buildMode == BuildMode.debug,
+      buildMode: buildMode,
       trackWidgetCreation: trackWidgetCreation,
       targetProductVm: buildMode == BuildMode.release,
```
Comment (alexmarkov) [inline packages/flutter_tools/lib/src/base/build.dart]: Maybe we can also get rid of 'targetProductVm' parameter.

File: packages/flutter_tools/lib/src/compile.dart [hunk 2]
```diff
@@ -238,6 +239,28 @@ class PackageUriMapper {
...
+        '--enable-asserts',
+      ];
+    case BuildMode.profile:
+      return <String>[
+        '-Ddart.vm.product=false',
+        '--bytecode-options=source-positions',
+      ];
+    case BuildMode.release:
+      return <String>[
+        '-Ddart.vm.product=true',
```
Comment (alexmarkov) [inline packages/flutter_tools/lib/src/compile.dart]: This is also set depending on 'targetProductVm' at lines 323-324. Consider cleaning up that code.
Also, consider defining '-Ddart.vm.profile=true' depending on the mode.

**Review by alexmarkov — APPROVED** [review]
LGTM

Maybe someone more familiar with flutter_tools should also take a look.

File: packages/flutter_tools/lib/src/base/build.dart (hunk 1 above)
Comment (rmacnak-google) [reply packages/flutter_tools/lib/src/base/build.dart]: Done.

File: packages/flutter_tools/lib/src/compile.dart (hunk 2 above)
Comment (rmacnak-google) [reply packages/flutter_tools/lib/src/compile.dart]: Done.

**Pull request merged**